File: ruby/sinatra/lib/controller/testing_controller.rb
# frozen_string_literal: true

# handles api testing setup
class TestingController
    def initialize(data_loader, storage)
        @data_loader = data_loader
        @storage = storage
    end

    def setup
        @storage.reset

        video_games = @data_loader.load

        video_games.each do |video_game|
            @storage.add video_game
        end
    end
end

File: js/express/lib/controllers/testing-controller.js
const DataLoader = require('../utils/data-loader');

function TestingController(storage) {
    this._storage = storage;
}

TestingController.prototype.setup = async function (res) {
    try {
        const videoGames = await DataLoader.load();

        this._storage.constructor();

        for (const videoGame of videoGames)
            this._storage.addVideoGame(videoGame);

        res.okText('Successfully loaded data.');
    } catch (error) {
        console.error(error);
        res.internalServerError();
    }
}

module.exports = TestingController;
